src/v/cloud_storage/topic_manifest.cc

```diff
@@ -12,21 +12,193 @@
 #include "cloud_storage/types.h"
+#include "cluster/types.h"
 #include "hashing/xx.h"
 #include "json/writer.h"
 #include "model/timestamp.h"
 
 #include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
 
 #include <boost/lexical_cast.hpp>
+#include <boost/lexical_cast/try_lexical_convert.hpp>
 #include <fmt/ostream.h>
+#include <rapidjson/encodings.h>
+#include <rapidjson/error/en.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/reader.h>
 
 #include <chrono>
+#include <optional>
+#include <stdexcept>
+#include <string_view>
 
 namespace cloud_storage {
+
+struct topic_manifest_handler
+  : public rapidjson::
+      BaseReaderHandler<rapidjson::UTF8<>, topic_manifest_handler> {
+    using key_string = ss::basic_sstring<char, uint32_t, 31>;
+    bool StartObject() {
+        switch (_state) {
+        case state::expectManifestStart:
+            _state = state::expectKey;
+            return true;
+        case state::expectKey:
+        case state::expectValue:
+            return false;
+        }
+    }
+
+    bool Key(const char* str, rapidjson::SizeType length, bool /*copy*/) {
+        switch (_state) {
+        case state::expectKey:
+            _key = key_string(str, length);
+            _state = state::expectValue;
+            return true;
+        case state::expectManifestStart:
+        case state::expectValue:
+            return false;
+        }
+    }
+
+    bool String(const char* str, rapidjson::SizeType length, bool /*copy*/) {
+        std::string_view sv(str, length);
+        switch (_state) {
+        case state::expectValue:
+            if ("namespace" == _key) {
+                _namespace = model::ns(ss::sstring(sv));
+            } else if ("topic" == _key) {
+                _topic = model::topic(ss::sstring(sv));
+            } else if ("compression" == _key) {
+                model::compression compression_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, compression_res)) {
+                    _properties.compression = compression_res;
+                } else {
+                    return false;
+                }
+            } else if ("cleanup_policy_bitflags" == _key) {
+                model::cleanup_policy_bitflags cleanup_policy_bitflags_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, cleanup_policy_bitflags_res)) {
+                    _properties.cleanup_policy_bitflags
+                      = cleanup_policy_bitflags_res;
+                } else {
+                    return false;
+                }
+            } else if ("compaction_strategy" == _key) {
+                model::compaction_strategy compaction_strategy_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, compaction_strategy_res)) {
+                    _properties.compaction_strategy = compaction_strategy_res;
+                }
```

Comment:
should there be an `else` case for when the conversion fails?

Reply:
It throws.

Reply:
I don't think try_lexical_convert throws. If it does it looks like the other usages in here would need updated too. Are you referring to something else?

Reply:
`boost::try_lexical_cast` uses `std::istream& operator>>(std::istream& i, compaction_strategy& cs)` which uses `string_switch`, which throws if there's not a match.

Reply:
I see. is it confusing it to have that in an `if statement`? it seems like it kinda circumvents the documentation for `try_lexical_convert` which makes the point that it returns true/false if the conversion worked/failed.

I wonder then if the other instances `try_lexical_convert` that use the `if (try_lexical_convert) else { return false; }`?

Reply:
changed it to `boost::lexical_cast` inside `try` in `topic_manifest::update`, after parsing is done for better error messages